exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetBooleanPredicate.java

```diff
@@ -46,15 +46,29 @@ private ParquetBooleanPredicate(String name, List<LogicalExpression> args, Expre
       ExpressionPosition pos
   ) {
     return new ParquetBooleanPredicate<C>(name, args, pos) {
+      /**
+       * Evaluates a compound "AND" filter on the statistics of a RowGroup (the filter reads "filterA and filterB").
+       * Return value :<ul>
+       *   <li>ALL : only if all filters return ALL
+       *   <li>NONE : if one filter at least returns NONE
+       *   <li>SOME : all other cases
+       * </ul>
+       */
       @Override
-      public boolean canDrop(RangeExprEvaluator<C> evaluator) {
-        // "and" : as long as one branch is OK to drop, we can drop it.
+      public RowsMatch matches(RangeExprEvaluator<C> evaluator) {
+        RowsMatch resultMatch = RowsMatch.ALL;
         for (LogicalExpression child : this) {
-          if (child instanceof ParquetFilterPredicate && ((ParquetFilterPredicate)child).canDrop(evaluator)) {
-            return true;
+          if (child instanceof ParquetFilterPredicate) {
+            switch(((ParquetFilterPredicate) child).matches(evaluator)) {
+              case NONE :
```

Comment:
Remove space before `:`. Please correct here and below.

Reply:
Done